src/org/zaproxy/zap/extension/httpsinfo/MDialog.java

```diff
@@ -0,0 +1,327 @@
+		} else {
+			content.append(Constant.messages.getString("httpsinfo.general.achieveable.strength.heading")).append(" ")
+					.append(strengthString(server.getMaxStrength())).append(NEWLINE);
+		}
+		if (server.getBeastVuln() == null) {
+			content.append(Constant.messages.getString("httpsinfo.general.beast.status.heading")).append(" ")
+					.append(Constant.messages.getString("httpsinfo.general.pending.indicator"));
+		} else {
+			content.append(Constant.messages.getString("httpsinfo.general.beast.status.heading")).append(" ")
+					.append(server.getBeastVuln()
+							? Constant.messages.getString("httpsinfo.general.vulnerable.indicator")
+							: Constant.messages.getString("httpsinfo.general.protected.indicator"))
+					.append(NEWLINE);
+			if (server.getBeastVuln()) {
+				raiseBeastAlert();
+			}
+		}
+		if (server.getCrimeVuln() == null) {
+			content.append(Constant.messages.getString("httpsinfo.general.crime.status.heading"))
+					.append(Constant.messages.getString("httpsinfo.general.pending.indicator"));
+		} else {
+			content.append(Constant.messages.getString("httpsinfo.general.crime.status.heading")).append(" ")
+					.append(server.getCrimeVuln()
+							? Constant.messages.getString("httpsinfo.general.vulnerable.indicator")
+							: Constant.messages.getString("httpsinfo.general.protected.indicator"));
+			if (server.getCrimeVuln()) {
+				raiseCrimeAlert();
+			}
+		}
+
+		general.setText(content.toString());
+	}
+
+	public void showCipherSuites() {
+		StringBuilder cs = new StringBuilder(Constant.messages.getString("httpsinfo.ciphersuites.supported.label"))
+				.append(" ").append(suppVers.getSelectedItem()).append(" ")
+				.append(ordered ? Constant.messages.getString("httpsinfo.ciphersuites.ordered")
+						: Constant.messages.getString("httpsinfo.ciphersuites.unordered"));
+		int x = 0;
+		try {
+			for (int c : server.getSupportedCipherSuites().get(versionStrings.get(suppVers.getSelectedItem()))) {
+				String entry = ordered ? x + "." : "  ";
+				cs.append(entry).append("      ").append(SSLServer.cipherSuiteString(c)).append(NEWLINE);
+				x++;
+			}
+
+		} catch (Exception e) {
+			cs.append("Pending...");
+			new Thread(new Runnable() {
+				@Override
+				public void run() {
+					server.updateSupportedCS();
+					server.updateAttackVuln();
+					server.updateStrength();
+					showGeneral();
+					showCipherSuites();
+				}
+			}).start();
+		}
+		cipherSuites.setText(cs.toString());
+	}
+
+	public void addComponentsToPane(Container pane) {
+		pane.setLayout(new GridBagLayout());
+		GridBagConstraints c = new GridBagConstraints();
+		c.fill = GridBagConstraints.BOTH;
+		c.gridwidth = 3;
+		c.anchor = GridBagConstraints.NORTH;
+		
+		c.weighty = 0.5;
+		c.gridx = 0;
+		c.gridy = 0;
+		pane.add(new JScrollPane(general), c);
+		 
+		c.gridwidth = 1;
+		c.weightx = 0.5;
+		c.weighty = 0;
+		c.gridx = 0;
+		c.gridy = 1;
+		pane.add(spinnerPref, c);
+
+		c.weightx = 0.5;
+		c.weighty = 0;
```

Comment:
This can be just set once, it uses the values previously set. (same comment for following "y=0" change.)